File: frontend/src/services/driverService.ts
const API_URL = import.meta.env.VITE_API_URL;

export async function getDrivers(
  raceName: string
) {
  const response = await fetch(
    `${API_URL}/api/v1/drivers?race=${raceName}`
  );

  if (!response.ok) {
    throw new Error("Failed to load drivers");
  }

  return response.json();
}

File: api/app/services/driver_service.py
import fastf1


def get_drivers(race_name: str):

    session = fastf1.get_session(
        2025,
        race_name,
        "R"
    )

    session.load()

    results = session.results

    return {
        "drivers": results["Abbreviation"].tolist()
    }